### PR TITLE
adding optional string auto-linking

### PR DESCRIPTION
**Pull request by amamparo**

### CODE REVIEW

**Comment by mac-s-g** [comment]
thanks for contributing!  I can review this week.

**Comment by JunlinZhu-Tommy** [comment]
We don't need package.json lock file and dist folder in this PR and should add them into gitignore.

File: .gitignore [hunk 1]
```diff
@@ -2,8 +2,7 @@
 npm-debug.log
 
 /coverage
-/dist
```
Comment (JunlinZhu-Tommy) [inline .gitignore]: Why we remove dist folder from here?

Comment (amamparo) [reply .gitignore]: Not sure, but you're right, `dist` should not be versioned. Updated pull request.

**Comment by dospolov** [comment]
@mac-s-g can you implement this functionality?

**Pull request closed without merging**